src/messages/message.ts

```diff
@@ -0,0 +1,49 @@
+import type { JsonMessage } from './types';
+
+import lodash from 'lodash';
+import { validate } from '../validation/validator';
+
+const { cloneDeep, isPlainObject } = lodash;
+export abstract class Message {
```

Comment:
Reminder that you wanted to remove the `abstract`

Reply:
currently leaning towards keeping it abstract because we don't to expose the ability to create instances of this class. thoughts?

Reply:
I was actually thinking to get rid of this abstract class entirely and just pass around `MessageSchema` directly.

Reply:
hmm. will we still have classes with `create` methods to simplify creation of specific `MessageSchema` types?

Reply:
You are right that classes will still have specific `create` methods that is not part of the abstract class. So in my view no value for this abstraction except calling `super(...)`, but that's just trading one line for another line of code at the moment.

Reply:
are you cool with us revisiting after we implement a few more interfaces? wondering if there may be some method we end up realizing needs to be implemented by each concrete message class

Reply:
Cool with that.